File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int count0 = 0;
        int count1 = 0;
        int count2 = 0;

        int n = nums.length;

        for(int i=0; i<n; i++) {
            if(nums[i] == 0) count0++;
            else if(nums[i] == 1) count1++;
            else count2++;
        }

        int index = 0;

        while(count0 != 0) {
            nums[index] = 0;
            index++;
            count0--;
        }

        while(count1 != 0) {
            nums[index] = 1;
            index++;
            count1--;
        }

        while(count2 != 0) {
            nums[index] = 2;
            index++;
            count2--;
        }
    }
}

File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int n = nums.size();

        int in = 0;
        for(int i=0; i<n; i++) {
            if(nums[i] == 0) {
                swap(nums[i], nums[in]);
                in++;
            }
        }

        for(int i=in; i<n; i++) {
            if(nums[i] == 1) {
                swap(nums[i], nums[in]);
                in++;
            }
        }
    }
};